### PR TITLE
tutorial: add tutorials on how to use iTowns

### LINKED ISSUE #1246: Create-a-simple-globe issue
Hello 
Apparently, this tutorial needs to be updated.
http://www.itowns-project.org/itowns/docs/#tutorials/Create-a-simple-globe
 
---

>` var position = new itowns.Coordinates('WGS84', 2.35, 48.8, 25e6); `

`var position = new itowns.Coordinates('EPSG:4326', 2.35, 48.8, 25e6);`

---

>var orthoSource = new itowns.WMTSSource({
```
var orthoSource = new itowns.WMTSSource({
projection: "EPSG:3857",
```
---
>var elevationSource = new itowns.WMTSSource({
name: 'ELEVATION.ELEVATIONGRIDCOVERAGE',
```

var elevationSource = new itowns.WMTSSource({
name: 'ELEVATION.ELEVATIONGRIDCOVERAGE.SRTM3',   <-- SRTM3 (?)
projection: "EPSG:4326",
```

### LINKED ISSUE #1304: Tutorials in the documentation
One thing that iTowns currently lacks is tutorials. There are only two at the
moment, and they are not even up-to-date. So here is a list of tutorials that I
think should be added to the documentation:

- [ ] how to display a vector tiles layer, and manipulate its style and filter
  the data
- [ ] how to pick things from layers containing `Feature`
- [ ] how to write a new parser and how the `Feature` system works
- [x] how to create a globe with a simple `ColorLayer` and `ElevationLayer`
  (already existing, needs some update)
- [x] how to display a `GeometryLayer`, and the explanation of its parameters
  (already existing, needs some update)

It is not much, but it could provide resources for a beginner at iTowns, without
having to look through each example and spend time reading the documentation.

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Add the following tutorial in iTowns documentation :

- What are the basics of iTowns (some generalities on how iTowns displays geographic data).
- How to display data in a local CRS (i.e. on a `PlanarView`).
- How to display vector data projected on the ground (i.e. vector data displayed as a `ColorLayer`).

Update the two already existing tutorials : 

- How to display data in a global CRS (i.e. on a `GlobeView`).
- How to display vector data as 3D objects (i.e. vector data displayed as a `GeometryLayer`).

Edit the documentation template to fix error with internal links to tutorials and to add support for tutorial sections.

## Motivation and Context

Improve iTowns documentation, closes #1246 and partialy answer the request in #1304.

## Left to do in this PR : 

- [x] fix internal links in tutorials

### CODE REVIEW

File: docs/tutorials/basics.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+On this matter, iTowns distinguishes two cases : 
+
+- The first one regards data that are to be displayed as raster. 
+  These data can either be original raster data or vector data whose representation is projected on the ground. 
+  Data from this type are displayed in a `GlobeView` if their source CRS is [WGS 84](https://epsg.io/4326) or [Pseudo-Mercator](https://epsg.io/3857).
+  However, if their source CRS defines a local projection (such as [RGF93 / Lambert 93](https://epsg.io/2154) for instance), they are displayed in a `PlanarView`.
+- On another hand, vector data that are to be displayed as 3D objects can be displayed in any type of `View`, regardless of their source CRS.
+
+It is important to aknowledge the following facts regarding original raster data and `Views` :
+- a `GlobeView` allows displaying multiple raster data from different source CRS,
```
Comment (gchoqueux) [inline docs/tutorials/basics.md]: GlobeView: 2 crs on raster : WGS84 and PM

File: docs/tutorials/basics.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+***
+
+## The support for visualized data
+
+It was earlier mentioned that data are displayed as `Layers` within iTowns. 
+Several specific types of `Layers` exist, the use of which depends on the data to display :
+
+- `ColorLayer` can be used to display raster graphics or vector data projected on the ground (left picture in the table bellow),
+- `ElevationLayer` can be used to display 3D elevation models (center picture in the table bellow),
+- `GeometryLayer` can be used to display 3D objects, such as geometric shapes or buildings modelling (right picture in the table bellow).
```
Comment (gchoqueux) [inline docs/tutorials/basics.md]: add `FeatureGeometryLayer`

File: docs/tutorials/basics.md (hunk 1 above)
Comment (mgermerie) [reply docs/tutorials/basics.md]: I modified the line.

File: docs/tutorials/basics.md (hunk 2 above)
Comment (mgermerie) [reply docs/tutorials/basics.md]: I added `FeatureGeometryLayer`.

**Review by gchoqueux — APPROVED** [review]
Thanks!

File: docs/tutorials/Fundamentals.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
+ITowns is a javascript framework for 3D geographic data visualisation. 
+It can display a wide range of data such as ortho-images, Digital Elevation Models (DEM) or 3D models.
+
+ITowns is based on [Three.js](https://threejs.org/), which is a javascript library that implements WebGL to render sophisticated 3D Geometry on a webpage.
+
+In this tutorial, we shall be introduced to the fundamentals of iTowns : how it displays geographic data.
+
+***
+
+## Data visualization with iTowns
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:10]: replace by

### The View

File: docs/tutorials/Fundamentals.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+It is important to aknowledge the following facts regarding original raster data and `Views` :
+- a `GlobeView` allows displaying multiple raster data from two different source CRS : [WGS 84](https://epsg.io/4326) and [Pseudo-Mercator](https://epsg.io/3857),
+- a `PlanarView` allows displaying multiple raster data, but all those data sources must have the same CRS.
+
+Tutorials of how to create and use the two `View` types can be found [here]{@tutorial Raster-data-WGS84} for the `GlobeView` and [here]{@tutorial Raster-data-Lambert93} for the `PlanarView`.
+
+***
+
+## The data supported by iTowns
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:37]: Replace by
### Protocols and data formats

File: docs/tutorials/Fundamentals.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+Each `View` in iTowns displays 3D data in a unique Coordinates Reference System (CRS). 
+Yet, the data displayed within a `View` do not necessarily have to be in the `View`'s CRS.
+ITowns comes with two pre-made `View` types : `GlobeView` and `PlanarView`. 
+Each of these allows visualizing data from different types (raster or vector) and from different CRS.
+On this matter, iTowns distinguishes two cases : 
+
+- The first one regards data that are to be displayed as raster. 
+  These data can either be original raster data or vector data whose representation is projected on the ground. 
+  Data from this type are displayed in a `GlobeView` if their source CRS is [WGS 84](https://epsg.io/4326) or [Pseudo-Mercator](https://epsg.io/3857).
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:25]: add documentation link to all itowns `Class`

File: docs/tutorials/Fundamentals.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+Regarding data formats, iTowns offers several possibilities : [vector tile](https://docs.mapbox.com/help/glossary/vector-tiles/) resources from [MapBox](https://www.mapbox.com/), [Potree](https://github.com/potree/potree) 3D point clouds, oriented images, [GeoJSON](https://geojson.org/), [KML](https://www.ogc.org/standards/kml) or [GPX](https://www.topografix.com/gpx.asp).
+
+***
+
+## The support for visualized data
+
+It was earlier mentioned that data are displayed as `Layers` within iTowns. 
+Several specific types of `Layers` exist, the use of which depends on the data to display :
+
+- `ColorLayer` can be used to display raster graphics or vector data projected on the ground (left picture in the table bellow),
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:53]: add documentation link to all itowns `Class`

File: docs/tutorials/Fundamentals.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+- `OrientedImageLayer` can be used to display some oriented images.
+
+
+| ![color layer](images/Fundamentals-1.png) | ![elevation layer](images/Fundamentals-2.png) | ![geometry layer](images/Fundamentals-3.png) |
+| :---: | :---: | :---: |
+| A simple `ColorLayer` is displayed | An `ElevationLayer` is added to represent terrain elevation | A `GeometryLayer` is added to model the buildings |
+
+***
+
+## Vector data appearance
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:67]: ### Style

File: docs/tutorials/Fundamentals.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+ITowns comes with a wide range of compatible data sources. 
+It can be used to display data from web servers that use protocols such as Web Map (Tile) Service ([WMS](https://www.ogc.org/standards/wms) and [WMTS](https://www.ogc.org/standards/wmts)), Web Feature Service ([WFS](https://www.ogc.org/standards/wfs)) and Tile Map Service ([TMS](https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification)).
+Data from user given files can also be displayed.
+
+Regarding data formats, iTowns offers several possibilities : [vector tile](https://docs.mapbox.com/help/glossary/vector-tiles/) resources from [MapBox](https://www.mapbox.com/), [Potree](https://github.com/potree/potree) 3D point clouds, oriented images, [GeoJSON](https://geojson.org/), [KML](https://www.ogc.org/standards/kml) or [GPX](https://www.topografix.com/gpx.asp).
+
+***
+
+## The support for visualized data
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:48]: ### Layers

File: docs/tutorials/Fundamentals.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+The content of the label will be the content stored in the data under the `name` property.
+
+### Vector data projected on the ground
+
+When vector data are flattened on the ground, they are displayed in a `ColorLayer`. 
+In that case, the data basically consist in polygons, lines or points.
+Their appearance can be adjusted by modifying the `Style` of the `ColorLayer`. 
+Yet, their positioning can't be modified since it is computed so that tey appear projected on the ground.
+
+Tutorial on how to modify the appearance of some vector data displayed on the ground can be found [here]{@tutorial Vector-data-on-ground}.
```
Comment (gchoqueux) [inline docs/tutorials/Fundamentals.md:106]: > Tutorial on how to modify the appearance of some vector data displayed on the ground can be found [here]{@tutorial Vector-data-on-ground}.

replace by 

:fast_forward: [**`Visiting Tutorial`**](http://www.itowns-project.org/itowns/docs/tutorials/Vector-data-on-ground.html)

File: docs/tutorials/Fundamentals.md (hunk 3 above)
Comment (mgermerie) [reply docs/tutorials/Fundamentals.md:10]: I changed it in PR #1679.

**Pull request merged**